crates/subspace-fraud-proof/src/tests.rs

```diff
@@ -91,6 +96,210 @@ impl VerifierApi for TestVerifierClient {
 // Use the system domain id for testing
 const TEST_DOMAIN_ID: DomainId = DomainId::new(3u32);
 
+type HashFor<Block> = <<Block as BlockT>::Header as HeaderT>::Hash;
+
+fn generate_storage_proof_for_tx_validity(
+    block_hash: HashFor<Block>,
+    client: Arc<DomainClient>,
+    keys: Vec<Vec<u8>>,
+) -> StorageProof {
+    client
+        .read_proof(
+            block_hash,
+            &mut keys
+                .iter()
+                .map(|k| k.as_slice())
+                .collect::<Vec<&[u8]>>()
+                .into_iter(),
+        )
+        .unwrap()
+}
+
+// TODO: Enable this test when runtime_api_light is fixed
```

Comment:
@vedhavyas I am getting the runtime code by calling the same runtime api in the test. Me and @NingLin-P both have encountered it before. 

 Can you point me to unit test where you are using the `derive_domain_timestamp_extrinsic`? It will help me see if I am doing something wrong.

Reply:
@vedhavyas From what I understand, The issue is encountered when we try to read `RuntimeVersion`. What @NingLin-P did in his PR was used `RuntimeVersionOf` constraint for `CodeExecutor` to read it from the executor itself rather than making call to `sc_executor::read_embedded_version`. 

I will talk with @NingLin-P and see if I can open an independent PR with the fix.